.circleci/config.yml

```diff
@@ -174,285 +250,178 @@ jobs:
+          release: buster
+          arch: aarch64
+          image: multiarch/debian-debootstrap:arm64-buster-slim
+          ext: deb
+          <<: *pkg_req
+      - package:
+          name: x64-debian-buster
+          dist: debian
+          release: buster
+          arch: x64
+          image: debian:buster
+          ext: deb
+          <<: *pkg_req
+      - package:
+          name: aarch64-debian-stretch
+          dist: debian
+          release: stretch
+          arch: aarch64
+          image: multiarch/debian-debootstrap:arm64-stretch-slim
+          ext: deb
+          <<: *pkg_req
+      - package:
+          name: x64-debian-stretch
+          dist: debian
+          release: stretch
+          arch: x64
+          image: debian:stretch
+          ext: deb
+          <<: *pkg_req
+      - package:
+          name: aarch64-centos-7
           dist: centos
           release: "7"
+          arch: aarch64
+          image: multiarch/centos:7-aarch64-clean
+          ext: rpm
           <<: *pkg_req
-      - build_rpms:
-          name: build_centos_8
+      - package:
+          name: x64-centos-7
+          dist: centos
+          release: "7"
+          arch: x64
+          image: centos:7
+          ext: rpm
+          <<: *pkg_req
+# there is no multiarch image for aarch64
+#      - package:
+#          name: aarch64-centos-8
+#          dist: centos
+#          release: "8"
+#          arch: aarch64
+#          image:
+#          ext: rpm
+#          <<: *pkg_req
+      - package:
+          name: x64-centos-8
           dist: centos
           release: "8"
+          arch: x64
+          image: centos:8
+          ext: rpm
+          <<: *pkg_req
+      - package:
+          name: x64-alpine-3
+          dist: alpine
+          release: "3"
+          arch: x64
+          image: alpine:3
+          ext: apk
+          workspace_extra: "/x86_64"
           <<: *pkg_req
-      - build_apks:
-          name: build_alpine
+      - package:
+          name: aarch64-alpine-3
+          dist: alpine
+          release: "3"
+          arch: aarch64
+          image: multiarch/alpine:aarch64-edge
+          ext: apk
+          workspace_extra: "/aarch64"
           <<: *pkg_req
       - collect_packages:
           requires:
-            - build_debian_stretch
-            - build_debian_buster
-            - build_ubuntu_xenial
-            - build_ubuntu_bionic
-            - build_centos_7
-            - build_centos_8
-            - build_alpine
+            - x64-ubuntu-xenial
+            - x64-ubuntu-bionic
+            - aarch64-ubuntu-xenial
+            - aarch64-ubuntu-bionic
+            - x64-debian-bullseye
+            - aarch64-debian-bullseye
+            - x64-debian-buster
+            - aarch64-debian-buster
+            - x64-debian-stretch
+            - aarch64-debian-stretch
+            - x64-centos-7
+            - x64-centos-8
+            - aarch64-centos-7
+#            - aarch64-centos-8
```

Comment:
`centos:8` doesn't work?

Reply:
No. It should be an ARM64 image. `centos:8` is the x86_64 one.
I am experimenting now with `--persistent yes` flag of multiarch/qemu-user-static. With this flag I was able to run `docker run -it --rm arm64v8/centos:8 uname -m` here. But step `Activate QEMU` fails at CircleCI with:
```
docker run -it --rm --privileged multiarch/qemu-user-static --reset --credential yes --persistent yes
Unable to find image 'multiarch/qemu-user-static:latest' locally
latest: Pulling from multiarch/qemu-user-static

aa22dec6: Pulling fs layer 
7f1a5b79: Pulling fs layer 
5179ae11: Pulling fs layer 
9da9c97d: Pulling fs layer 
Digest: sha256:54ed5674b4d96a2200a75373993a48d8c5dc3433ed4cdeef95a0e947e88bb05f
Status: Downloaded newer image for multiarch/qemu-user-static:latest
^@^@Setting /usr/bin/qemu-alpha-static as binfmt interpreter for alpha
sh: write error: Invalid argument
Setting /usr/bin/qemu-arm-static as binfmt interpreter for arm
sh: write error: Invalid argument
```

https://app.circleci.com/pipelines/github/martin-g/varnish-cache/198/workflows/74a620b9-d05e-4ec7-a24b-10e579225bac/jobs/1192

Reply:
I've found the problem. It was the CircleCI `machine` executor uses Ubuntu 14.04 by default and kernel 4.4 which is too old for QEMU. I've changed it to Ubuntu 16.04 with 4.14 and now I use Docker images produced by `arm64v8` DockerHub organization, not from `multiarch` org.